src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/ConcurrencyLimiter.cs

```diff
@@ -27,14 +27,12 @@ public sealed class ConcurrencyLimiter : RateLimiter
         private static readonly ConcurrencyLease SuccessfulLease = new ConcurrencyLease(true, null, 0);
         private static readonly ConcurrencyLease FailedLease = new ConcurrencyLease(false, null, 0);
         private static readonly ConcurrencyLease QueueLimitLease = new ConcurrencyLease(false, null, 0, "Queue limit reached");
-        private static readonly double TickFrequency = (double)TimeSpan.TicksPerSecond / Stopwatch.Frequency;
 
         // Use the queue as the lock field so we don't need to allocate another object for a lock and have another field in the object
         private object Lock => _queue;
 
         /// <inheritdoc />
-        public override TimeSpan? IdleDuration => _idleSince is null ? null : new TimeSpan((long)((Stopwatch.GetTimestamp() - _idleSince) * TickFrequency));
-
+        public override TimeSpan? IdleDuration => _idleSince is null ? null : RateLimiterHelper.GetElapsedTime(_idleSince.Value);
```

Comment:
Does every call site follow the same pattern of being:
```C#
nullableTimeSpan is null ? null : GetElapsedTime(nullableTimeSpan.Value)
```
? If so, should the helper do that whole thing to avoid the duplication at each call site?

Reply:
Good point, thanks